library/src/main/java/com/otaliastudios/zoom/internal/movement/ZoomManager.kt

```diff
@@ -115,13 +129,23 @@ internal class ZoomManager(provider: () -> MatrixController) : MovementManager(p
         var minZoom = getMinZoom()
         var maxZoom = getMaxZoom()
         if (allowOverZoom && isOverEnabled) {
-            minZoom -= maxOverZoom
-            maxZoom += maxOverZoom
+            minZoom -= maxOverZoomOut
+            maxZoom += maxOverZoomIn
         }
         return value.coerceIn(minZoom, maxZoom)
     }
 
     companion object {
-        private const val DEFAULT_OVERZOOM_FACTOR = 0.1f
+        const val DEFAULT_OVERZOOM_FACTOR = 0.1f
+        val DEFAULT_OVERZOOM_PROVIDER = object : OverZoomRangeProvider {
+            override fun getOverZoomIn(engine: ZoomEngine): Float {
+                return DEFAULT_OVERZOOM_FACTOR * (engine.getMaxZoom() - engine.getMinZoom())
+            }
+
+            override fun getOverZoomOut(engine: ZoomEngine): Float {
+                return DEFAULT_OVERZOOM_FACTOR * (engine.getMaxZoom() - engine.getMinZoom())
+            }
+        }
```

Comment:
Can we move this to OverZoomRangeProvider interface so people can call `OverZoomRangeProvider.DEFAULT`? Also I don't think the factor should be public (it's not like a carefully tuned value) but if you need it public, let's make the pan factor public too.